vizro-core/src/vizro/static/js/actions/build_action_loop_callbacks.js

```
@@ -25,7 +25,10 @@ function gateway(
   ) {
     // If there's no more actions to execute, stop the loop perform.
     if (remaining_actions.length == 0) {
-      return dash_clientside.PreventUpdate;
+      // Return list of dash.no_update for all outputs to circumvent the JS "RangeError" exception for the dash==3.0.4
```

I think we can do `throw dash_clientside.PreventUpdate` like Python's `raise PreventUpdate` rather than all these `return dash.no_update`?

---

You're right! All errors disappear when `return dash_clientside.PreventUpdate` is replaced with the `throw dash_clientside.PreventUpdate`